### PR TITLE
[DOCS] Expands classification documentation with multiclass

### PR DESCRIPTION
**Pull request by szabosteve**
## Overview

This PR expands the classification documentation to refer to multiclass classification.
New elements:
* maximum number of classes
* reworked section about imbalanced class sizes
* adjusted description of the classification algorithms
* section on classification performance
* adjusted section on evaluation in the classification conceptual piece
* two examples with images of confusion matrices in the evaluate piece

## Previews

* [Classification](http://stack-docs_973.docs-preview.app.elstc.co/guide/en/machine-learning/master/dfa-classification.html)
* [Evaluation](http://stack-docs_973.docs-preview.app.elstc.co/guide/en/machine-learning/master/ml-dfanalytics-evaluate.html)

### CODE REVIEW

File: docs/en/stack/ml/df-analytics/dfa-classification.asciidoc [hunk 1]
```diff
@@ -49,11 +41,25 @@ means that you need to supply a labeled training dataset that has some
...
-Your training dataset should be approximately balanced which means the number of 
-data points belonging to the various classes should not be widely different, 
-otherwise the {classanalysis} may not provide the best predictions. Read 
-<<dfa-classification-imbalanced-classes>> to learn more.
+learned about the relationships between the data points to classify new data. In 
+case there are more than two classes, an automated process called stratification 
+makes sure that the proportions of each class in the training data are equal so 
+that the training is consistent. 
+
+Your training dataset should be approximately balanced. It means the number of
```
Comment (lcawl) [inline docs/en/stack/ml/df-analytics/dfa-classification.asciidoc:49]: Minor detail: "dataset" is two words in the style guide
```suggestion
Your training data set should be approximately balanced. It means the number of 
```

File: docs/en/stack/ml/df-analytics/dfa-classification.asciidoc [hunk 2]
```diff
@@ -64,10 +70,33 @@ otherwise the {classanalysis} may not provide the best predictions. Read
...
+previous one. Every tree is an iteration of the last one, hence it improves the 
+decision made by the previous tree.
 //end::classification-algorithms[]
 
 
+[discrete]
+[[dfa-classification-performance]]
+==== {classification-cap} performance
+
+As a rule of thumb, a {classanalysis} with many classes takes more time to run
```
Comment (lcawl) [inline docs/en/stack/ml/df-analytics/dfa-classification.asciidoc:84]: > As a rule of thumb...

This sentence seems like it should be at the start of the next paragraph, not this one. This paragraph seems to be talking mostly about the number of documents, whereas the next one is talking about the number of classes.

File: docs/en/stack/ml/df-analytics/dfa-classification.asciidoc [hunk 3]
```diff
@@ -37,23 +29,37 @@ these fields as _{feature-vars}_ and _{depvar}_, respectively.
 {feature-vars-cap} are the values that the {depvar} value depends on. There are 
 three different types of {feature-vars} that you can use with our 
 {classification} algorithm: numerical, categorical, and boolean. Arrays are not 
-supported in the {feature-var} fields.
+supported in the {feature-var} fields.
```
Comment (lcawl) [inline docs/en/stack/ml/df-analytics/dfa-classification.asciidoc:32]: > Feature variables

This can perhaps be pursued in a separate PR, but it occurred to me while reading this page that neither our UI nor the API refer to "feature variables". They refer to a "dependent variable" and to included or excluded "analyzed fields". Should we therefore remove this section label and term? Or make it less prominent?

I think this section boils down to something like this: When you create a classification job, you must specify which field contains the classification labels/values/categories that you want to predict (no more than 30). You can also optionally specify which fields to include or exclude from the analysis.  [It might be nice to mention here why you would choose to explicitly include/exclude fields as opposed to letting the analytics do it for you. Why bother? And also maybe mention how you can use the explain API to see which fields are used]

File: docs/en/stack/ml/df-analytics/dfa-classification.asciidoc [hunk 4]
```diff
@@ -127,18 +157,20 @@ include::{docdir}/shared-ml-concepts.asciidoc[tag=feature-importance]
...
-evaluation provides you is the multi-class confusion matrix which tells you how 
-many times a given data point that belongs to a given class was classified 
-correctly and incorrectly. In other words, how many times your data point that 
-belongs to the X class was mistakenly classified as Y.
+evaluation provides you is a confusion matrix. The more classes you have, the 
+more complex the confusion matrix is. The matrix tells you how many data points 
+that belong to a given class were classified correctly and incorrectly. In other 
+words, how many data points that belong to the X class were classified correctly 
+and mistakenly classified as Y or Z or any other class in the data set other than 
+X.
```
Comment (lcawl) [inline docs/en/stack/ml/df-analytics/dfa-classification.asciidoc:168]: > In other words, how many data points...

In my opinion, this sentence is redundant (i.e. doesn't add anything to the previous statement). What I think would be more interesting to add (in a separate PR if necessary) is what to do about poor classification results.  Can we provide tips about whether to improve it by re-running with feature importance enabled or using the explain API to determine which fields we should explicitly include or exclude? Or by curating the training data?

File: docs/en/stack/ml/df-analytics/dfa-classification.asciidoc (hunk 3 above)
Comment (szabosteve) [reply docs/en/stack/ml/df-analytics/dfa-classification.asciidoc]: @lcawl 
In 860f884, I made feature variables less prominent. I still use the term between brackets though, I felt it won't hurt to name it even if we don't use the term on the UI or in the API docs. If you think we should omit it entirely, we can easily delete it without further complications. Also mentioned why you would choose to include/exclude fields and mentioned the Explain API with a link. I also removed the details of supported fields, but put a link that points to the supported field-types in the PUT DFA API docs (which creates a dependency between this PR and the other creating the link).

File: docs/en/stack/ml/df-analytics/dfa-classification.asciidoc (hunk 4 above)
Comment (szabosteve) [reply docs/en/stack/ml/df-analytics/dfa-classification.asciidoc:173]: I deleted the sentence and address the rest in a separate PR.

**Comment by lcawl** [comment]
@elasticmachine run elasticsearch-ci/docs

File: docs/en/stack/ml/df-analytics/evaluatedf-api.asciidoc [hunk 5]
```diff
@@ -147,15 +147,37 @@ This evaluation type is suitable for evaluating {classification} models. In the
 classes. The {classification} evaluation offers the following metrics to
```
Comment (lcawl) [inline docs/en/stack/ml/df-analytics/evaluatedf-api.asciidoc:147]: > In the {version} version of the {stack}, the {classification} evaluation handles two classes.

I think this sentence is no longer true and should be removed.

File: docs/en/stack/ml/df-analytics/dfa-classification.asciidoc [hunk 6]
```diff
@@ -5,55 +5,51 @@
...
-{stack}, you can perform only binary {classanalysis}.
+{reganalysis} which predicts continuous, numerical values. The maximum number of 
+classes for {classanalysis} is 30.
 
 
 [discrete]
-[[dfa-classification-features]]
-==== {feature-vars-cap}
+[[dfa-classification-depvar]]
+==== {depvar-cap}
```
Comment (lcawl) [inline docs/en/stack/ml/df-analytics/dfa-classification.asciidoc:24]: > {depvar-cap}

This attribute doesn't exist in the shared attributes file, so it's not resolving.  I actually recommend omitting this section header and anchor.  I think this paragraph flows well from the one above it.

File: docs/en/stack/ml/df-analytics/dfa-classification.asciidoc [hunk 7]
```diff
@@ -5,55 +5,51 @@
...
 
-When you perform {classification}, you must identify a subset of fields that you 
-want to use to create a model for predicting another field value. We refer to 
-these fields as _{feature-vars}_ and _{depvar}_, respectively. 
-{feature-vars-cap} are the values that the {depvar} value depends on. There are 
-three different types of {feature-vars} that you can use with our 
-{classification} algorithm: numerical, categorical, and boolean. Arrays are not 
-supported in the {feature-var} fields.
+When you create a {classification} job, you must specify which field contains 
+the classes that you want to predict. We refer to this field as {depvar}. The
```
Comment (lcawl) [inline docs/en/stack/ml/df-analytics/dfa-classification.asciidoc:27]:
```suggestion
the classes that you want to predict. This field is known as the {depvar}. The 
```

File: docs/en/stack/ml/df-analytics/dfa-classification.asciidoc [hunk 8]
```diff
@@ -5,55 +5,51 @@
...
-three different types of {feature-vars} that you can use with our 
-{classification} algorithm: numerical, categorical, and boolean. Arrays are not 
-supported in the {feature-var} fields.
+When you create a {classification} job, you must specify which field contains 
+the classes that you want to predict. We refer to this field as {depvar}. The 
+rest of the fields that are included in the analysis are used to create the 
+model for predicting the value of the {depvar} (these are the {feature-vars}). 
+All {ref}/put-dfanalytics.html#dfa-supported-fields[supported fields] are 
+included by default, however, you can explicitly exclude and/or include fields. 
+Since the runtime and the necessary resources for the analysis are increasing as
```
Comment (lcawl) [inline docs/en/stack/ml/df-analytics/dfa-classification.asciidoc:32]:
```suggestion
Since the runtime and the necessary resources for the analysis increase as 
```

File: docs/en/stack/ml/df-analytics/dfa-classification.asciidoc [hunk 9]
```diff
@@ -5,55 +5,51 @@
...
-{classification} algorithm: numerical, categorical, and boolean. Arrays are not 
-supported in the {feature-var} fields.
+When you create a {classification} job, you must specify which field contains 
+the classes that you want to predict. We refer to this field as {depvar}. The 
+rest of the fields that are included in the analysis are used to create the 
+model for predicting the value of the {depvar} (these are the {feature-vars}). 
+All {ref}/put-dfanalytics.html#dfa-supported-fields[supported fields] are 
+included by default, however, you can explicitly exclude and/or include fields. 
+Since the runtime and the necessary resources for the analysis are increasing as 
+the number of included fields is growing, consider to include only relevant
```
Comment (lcawl) [inline docs/en/stack/ml/df-analytics/dfa-classification.asciidoc:33]:
```suggestion
the number of included fields grow, consider including only relevant 
```

File: docs/en/stack/ml/df-analytics/dfa-classification.asciidoc [hunk 10]
```diff
@@ -5,55 +5,51 @@
...
-data points belonging to the various classes should not be widely different, 
-otherwise the {classanalysis} may not provide the best predictions. Read 
-<<dfa-classification-imbalanced-classes>> to learn more.
+means that you need to supply a labeled training data set that has a {depvar} 
+and some fields that are related to it. The {classification} algorithm learns 
+the relationships between these fields and the {depvar}. Once you’ve trained the 
+model on your training data set, you can reuse the knowledge that the model has 
+learned about the relationships between the data points to classify new data.
+
+Although the effects of imbalanced data are automatically mitigated before the
```
Comment (lcawl) [inline docs/en/stack/ml/df-analytics/dfa-classification.asciidoc:49]: Not mandatory, but I think it would be more impactful to split this sentence.
```suggestion
The effects of imbalanced data are automatically mitigated before the 
```

File: docs/en/stack/ml/df-analytics/dfa-classification.asciidoc [hunk 11]
```diff
@@ -5,55 +5,51 @@
...
-otherwise the {classanalysis} may not provide the best predictions. Read 
-<<dfa-classification-imbalanced-classes>> to learn more.
+means that you need to supply a labeled training data set that has a {depvar} 
+and some fields that are related to it. The {classification} algorithm learns 
+the relationships between these fields and the {depvar}. Once you’ve trained the 
+model on your training data set, you can reuse the knowledge that the model has 
+learned about the relationships between the data points to classify new data.
+
+Although the effects of imbalanced data are automatically mitigated before the 
+training, we still recommend using an approximately balanced data set to train
```
Comment (lcawl) [inline docs/en/stack/ml/df-analytics/dfa-classification.asciidoc:50]: Another minor suggestion to remove "we" if possible:
```suggestion
training. Nonetheless, it is a good idea to train your model with a data set that is approximately balanced.
```

File: docs/en/stack/ml/df-analytics/dfa-classification.asciidoc [hunk 12]
```diff
@@ -5,55 +5,51 @@
...
-<<dfa-classification-imbalanced-classes>> to learn more.
+means that you need to supply a labeled training data set that has a {depvar} 
+and some fields that are related to it. The {classification} algorithm learns 
+the relationships between these fields and the {depvar}. Once you’ve trained the 
+model on your training data set, you can reuse the knowledge that the model has 
+learned about the relationships between the data points to classify new data.
+
+Although the effects of imbalanced data are automatically mitigated before the 
+training, we still recommend using an approximately balanced data set to train 
+the model. It means the number of data points belonging to the various classes
```
Comment (lcawl) [inline docs/en/stack/ml/df-analytics/dfa-classification.asciidoc:51]: I find when sentences start with "it", I have to go back to previous sentences to figure out what "it" is.  So I'd reword to something like this:
```suggestion
That is to say, ideally your data set should have a similar number of data points for each class.
```

File: docs/en/stack/ml/df-analytics/dfa-classification.asciidoc [hunk 13]
```diff
@@ -5,55 +5,51 @@
...
+means that you need to supply a labeled training data set that has a {depvar} 
+and some fields that are related to it. The {classification} algorithm learns 
+the relationships between these fields and the {depvar}. Once you’ve trained the 
+model on your training data set, you can reuse the knowledge that the model has 
+learned about the relationships between the data points to classify new data.
+
+Although the effects of imbalanced data are automatically mitigated before the 
+training, we still recommend using an approximately balanced data set to train 
+the model. It means the number of data points belonging to the various classes 
+is not widely different.
```
Comment (lcawl) [inline docs/en/stack/ml/df-analytics/dfa-classification.asciidoc:52]: If you merge the previous suggestions, this sentence fragment is not longer required:
```suggestion

```

File: docs/en/stack/ml/df-analytics/dfa-classification.asciidoc [hunk 14]
```diff
@@ -64,10 +60,32 @@ otherwise the {classanalysis} may not provide the best predictions. Read
...
 
+[discrete]
+[[dfa-classification-performance]]
+==== {classification-cap} performance
+
+As a rule of thumb, a {classanalysis} with many classes takes more time to run 
+than a binary {classification} process when there are only two classes. The 
+relationship between the number of classes and the runtime is linear.
+
+The runtime scales approximately linearly with the number of involved documents
```
Comment (lcawl) [inline docs/en/stack/ml/df-analytics/dfa-classification.asciidoc:78]:
```suggestion
The runtime also scales approximately linearly with the number of involved documents 
```

File: docs/en/stack/ml/df-analytics/dfa-classification.asciidoc [hunk 15]
```diff
@@ -64,10 +60,32 @@ otherwise the {classanalysis} may not provide the best predictions. Read
...
+As a rule of thumb, a {classanalysis} with many classes takes more time to run 
+than a binary {classification} process when there are only two classes. The 
+relationship between the number of classes and the runtime is linear.
+
+The runtime scales approximately linearly with the number of involved documents 
+below 200.000 data points. Therefore, if you double the number of documents, 
+then the runtime of the analysis doubles respectively. Above 200.000 data 
+points, runtime increases faster.
+
+To achieve faster runtime, consider to set a smaller training percent, prepare
```
Comment (lcawl) [inline docs/en/stack/ml/df-analytics/dfa-classification.asciidoc:83]: > To achieve faster runtime...

This sentence is similar to the previous sentence ("runtime increases faster"), but their meanings are different in a way that I find a bit confusing.  To avoid confusion, maybe we could change this sentence to something like:

> To improve the performance of your {classanalysis}....

File: docs/en/stack/ml/df-analytics/dfa-classification.asciidoc [hunk 16]
```diff
@@ -64,10 +60,32 @@ otherwise the {classanalysis} may not provide the best predictions. Read
...
+than a binary {classification} process when there are only two classes. The 
+relationship between the number of classes and the runtime is linear.
+
+The runtime scales approximately linearly with the number of involved documents 
+below 200.000 data points. Therefore, if you double the number of documents, 
+then the runtime of the analysis doubles respectively. Above 200.000 data 
+points, runtime increases faster.
+
+To achieve faster runtime, consider to set a smaller training percent, prepare 
+your input data to have less classes if it is possible, and remove the fields
```
Comment (lcawl) [inline docs/en/stack/ml/df-analytics/dfa-classification.asciidoc:84]: This sentence is quite long, so I'd recommend splitting it:
```suggestion
If possible, prepare your input data such that is has less classes. You can also remove the fields 
```

File: docs/en/stack/ml/df-analytics/dfa-classification.asciidoc [hunk 17]
```diff
@@ -64,10 +54,33 @@ otherwise the {classanalysis} may not provide the best predictions. Read
...
+As a rule of thumb, a {classanalysis} with many classes takes more time to run 
+than a binary {classification} process when there are only two classes. The 
+relationship between the number of classes and the runtime is linear.
+
+The runtime also scales approximately linearly with the number of involved 
+documents below 200.000 data points. Therefore, if you double the number of 
+documents, then the runtime of the analysis doubles respectively. Above 200.000 
+data points, runtime increases faster.
+
+To improve the performance of your {classanalysis}, consider to set a smaller
```
Comment (lcawl) [inline docs/en/stack/ml/df-analytics/dfa-classification.asciidoc:77]: > To improve the performance of your {classanalysis}, consider to set a smaller training percent. 

When I read this, it's not immediately obvious what this means.  I had to go look at the API for a definition. Could we maybe change it to something like this?:

> To improve... consider using a smaller `training_percent` value when you create the job. That is to say, use a smaller percentage of your documents to train the model more quickly.

File: docs/en/stack/ml/df-analytics/dfa-classification.asciidoc [hunk 18]
```diff
@@ -64,10 +54,33 @@ otherwise the {classanalysis} may not provide the best predictions. Read
 The ensemble algorithm that we use in the {stack} for {classification} is a type
```
Comment (lcawl) [inline docs/en/stack/ml/df-analytics/dfa-classification.asciidoc:54]: > The ensemble algorithm that we use....

This can likely be addressed in a separate PR, but I think we can rewrite these sentence to take "us" out of the picture.  For example, something like:  Classification analysis uses an ensemble algorithm... It uses decision trees...

File: docs/en/stack/ml/df-analytics/dfa-classification.asciidoc [hunk 19]
```diff
@@ -17,57 +17,72 @@ other genres available in the set of categories we have. Therefore,
...
+As a rule of thumb, a {classanalysis} with many classes takes more time to run 
+than a binary {classification} process when there are only two classes. The 
+relationship between the number of classes and the runtime is linear.
+
+The runtime also scales approximately linearly with the number of involved 
+documents below 200.000 data points. Therefore, if you double the number of 
+documents, then the runtime of the analysis doubles respectively. Above 200.000 
+data points, runtime increases faster.
+
+To improve the performance of your {classanalysis}, consider to set a smaller
```
Comment (lcawl) [inline docs/en/stack/ml/df-analytics/dfa-classification.asciidoc:78]: Minor wording change. I think it works better as "setting" or "using" than "to set".
```suggestion
To improve the performance of your {classanalysis}, consider using a smaller 
```

File: docs/en/stack/ml/df-analytics/dfa-classification.asciidoc [hunk 20]
```diff
@@ -17,57 +17,72 @@ other genres available in the set of categories we have. Therefore,
...
+
+The runtime also scales approximately linearly with the number of involved 
+documents below 200.000 data points. Therefore, if you double the number of 
+documents, then the runtime of the analysis doubles respectively. Above 200.000 
+data points, runtime increases faster.
+
+To improve the performance of your {classanalysis}, consider to set a smaller 
+`training_percent` value when you create the job. That is to say, use a smaller 
+percentage of your documents to train the model more quickly. If possible, 
+prepare your input data such that is has less classes. You can also remove the
```
Comment (lcawl) [inline docs/en/stack/ml/df-analytics/dfa-classification.asciidoc:81]: Small typo
```suggestion
prepare your input data such that it has less classes. You can also remove the 
```

File: docs/en/stack/ml/df-analytics/dfa-classification.asciidoc [hunk 21]
```diff
@@ -17,57 +17,72 @@ other genres available in the set of categories we have. Therefore,
...
+[discrete]
+[[dfa-classification-performance]]
+==== {classification-cap} performance
+
+As a rule of thumb, a {classanalysis} with many classes takes more time to run 
+than a binary {classification} process when there are only two classes. The 
+relationship between the number of classes and the runtime is linear.
+
+The runtime also scales approximately linearly with the number of involved 
+documents below 200.000 data points. Therefore, if you double the number of
```
Comment (lcawl) [inline docs/en/stack/ml/df-analytics/dfa-classification.asciidoc:74]: Our style guide recommends using a comma to separate out numbers larger than 999.
```suggestion
documents below 200,000 data points. Therefore, if you double the number of 
```

File: docs/en/stack/ml/df-analytics/dfa-classification.asciidoc [hunk 22]
```diff
@@ -17,57 +17,72 @@ other genres available in the set of categories we have. Therefore,
...
+[[dfa-classification-performance]]
+==== {classification-cap} performance
+
+As a rule of thumb, a {classanalysis} with many classes takes more time to run 
+than a binary {classification} process when there are only two classes. The 
+relationship between the number of classes and the runtime is linear.
+
+The runtime also scales approximately linearly with the number of involved 
+documents below 200.000 data points. Therefore, if you double the number of 
+documents, then the runtime of the analysis doubles respectively. Above 200.000
```
Comment (lcawl) [inline docs/en/stack/ml/df-analytics/dfa-classification.asciidoc:75]: Ditto re thousand separator:
```suggestion
documents, then the runtime of the analysis doubles respectively. Above 200,000 
```

File: docs/en/stack/ml/df-analytics/dfa-classification.asciidoc [hunk 23]
```diff
@@ -4,70 +4,100 @@
...
-want to use to create a model for predicting another field value. We refer to 
-these fields as _{feature-vars}_ and _{depvar}_, respectively. 
-{feature-vars-cap} are the values that the {depvar} value depends on. There are 
-three different types of {feature-vars} that you can use with our 
-{classification} algorithm: numerical, categorical, and boolean. Arrays are not 
-supported in the {feature-var} fields.
+{classification-cap} is a {ml} process that enables you to predict the class or
+category of a data point in your data set. Typical examples of {classification}
+problems are predicting loan risk, classifying music, or detecting cancer in a
+DNA sequence. In the first case, for example, the data set might contain the
```
Comment (Winterflower) [inline docs/en/stack/ml/df-analytics/dfa-classification.asciidoc:10]: Hello everyone! 👋 
Very minor issue, but I would consider rewording this sentence `detecting cancer in a
DNA sequence` to something like `detecting the potential for cancer from a DNA sequence` , because technically cancer is an illness in a tissue/organ and not in the DNA sequence.

File: docs/en/stack/ml/df-analytics/dfa-classification.asciidoc [hunk 24]
```diff
@@ -4,70 +4,100 @@
...
-otherwise the {classanalysis} may not provide the best predictions. Read 
-<<dfa-classification-imbalanced-classes>> to learn more.
-
+{classification-cap} – just like {regression} – is a supervised {ml} process.
+When you create the {dfanalytics-job}, you must provide a data set that contains
+the _ground truth_. That is to say, your data set must contain the {depvar} 
+and the {feature-vars} fields that are related to it. You can divide the data
+set into training and testing data by specifying a `training_percent`. By
+default when you use the
+{ref}/put-dfanalytics.html[create {dfanalytics-jobs} API], 100% of the eligible
```
Comment (Winterflower) [inline docs/en/stack/ml/df-analytics/dfa-classification.asciidoc:40]: Could we please clarify what `eligible` documents are? For ex, I know we exclude documents if they have a field that has an array with more than one element (cc: @dimitris-athanasiou - could you please assist in getting this worded correctly). We don't have to define it in this paragraph, but it would be nice to link to a section that explains what documents are eligible.

File: docs/en/stack/ml/df-analytics/dfa-classification.asciidoc [hunk 25]
```diff
@@ -4,70 +4,100 @@
...
+documents in the data set are used for training. If you divide your data set,
+the job stratifies the data to ensure that both the training and testing data
+sets are representative of the full data set. That is to say, the same
```
Comment (Winterflower) [inline docs/en/stack/ml/df-analytics/dfa-classification.asciidoc:43]: I would clarify the concept of stratification here by saying `the training and testing data contains classes in proportions that are representative of the class proportions in the full data set` and maybe take out the next sentence - the one starting with `That is to say,` (cc: @tveasey )

File: docs/en/stack/ml/df-analytics/dfa-classification.asciidoc [hunk 26]
```diff
@@ -4,70 +4,100 @@
...
+proportions of each class are fairly represented in both data sets.
+
+When you are collecting a data set to train your model, ensure that it
+captures information for all of the classes. If some classes are poorly
+represented in the training data set, the model might be unaware of them. In
```
Comment (Winterflower) [inline docs/en/stack/ml/df-analytics/dfa-classification.asciidoc:48]: I would clarify this part `If some classes are poorly
represented in the training data set` by saying something like `If some classes are poorly
represented in the training data set (that is, you have very few data points per class)`

File: docs/en/stack/ml/df-analytics/dfa-classification.asciidoc [hunk 27]
```diff
@@ -4,70 +4,100 @@
...
 //end::classification-algorithms[]
 
 
+[discrete]
+[[dfa-classification-performance]]
+==== {classification-cap} performance
+
+As a rule of thumb, a {classanalysis} with many classes takes more time to run 
+than a binary {classification} process when there are only two classes. The 
+relationship between the number of classes and the runtime is linear.
```
Comment (Winterflower) [inline docs/en/stack/ml/df-analytics/dfa-classification.asciidoc:86]: I would say the `runtime is roughly linear` here to allow ourselves some wiggle room.

File: docs/en/stack/ml/df-analytics/dfa-classification.asciidoc [hunk 28]
```diff
@@ -4,70 +4,100 @@
...
+
+The runtime also scales approximately linearly with the number of involved 
+documents below 200,000 data points. Therefore, if you double the number of
```
Comment (Winterflower) [inline docs/en/stack/ml/df-analytics/dfa-classification.asciidoc:89]: I think this paragraph might be a too much detail, because the exact conditions that we have when testing the runtime won't necessarily translate to the user's environment, so they might experience something different. I would leave in this sentence starting with `Therefore, if you double the number of  documents, ...`
I would remove this `Above 200,000 
data points, runtime increases faster.`, because it is a bit too specific.

File: docs/en/stack/ml/df-analytics/dfa-classification.asciidoc [hunk 29]
```diff
@@ -4,70 +4,100 @@
...
+relationship between the number of classes and the runtime is linear.
+
+The runtime also scales approximately linearly with the number of involved 
+documents below 200,000 data points. Therefore, if you double the number of 
+documents, then the runtime of the analysis doubles respectively. Above 200,000 
+data points, runtime increases faster.
+
+To improve the performance of your {classanalysis}, consider using a smaller 
+`training_percent` value when you create the job. That is to say, use a smaller 
+percentage of your documents to train the model more quickly. If possible,
```
Comment (Winterflower) [inline docs/en/stack/ml/df-analytics/dfa-classification.asciidoc:95]: I would add here that the key is iteration. The user can start by using a smaller training percentage, run the job, check the performance and then decide if it's necessary to increase the training percentage.

**Comment by szabosteve** [comment]
@elasticmachine test this please

**Comment by szabosteve** [comment]
**Backport commits:**
7.x: 091a640
7.7: ef5022b
